pkg/standalone/standalone.go

```diff
@@ -225,8 +226,7 @@ func makePodSourceConfig(kc *KubeletConfig) *config.PodConfig {
 		glog.Infof("Adding manifest url: %v", kc.ManifestURL)
 		config.NewSourceURL(kc.ManifestURL, kc.HttpCheckFrequency, cfg.Channel(kubelet.HTTPSource))
 	}
-
-	if kc.EtcdClient != nil {
+	if !reflect.ValueOf(kc.EtcdClient).IsNil() {
```

Comment:
Hmmm, so what you're saying is that `kubelet.EtcdClientOrDie()` should return type `tools.EtcdClient` (the interface) and not `*etcd.Client` (pointer to the real type) is that it? In that case the `return nil` there would create a real `nil` that could be compared here without using the `reflect` package, is that correct?

On the other hand, using `reflect` here works, so is there an advantage to having it be a real `nil` instead of a value that evaluates to `nil` under reflection?